tests/workflows/conftest.py

```
@@ -5,36 +5,196 @@
+
+
+_ACT_RUNTIME_STATUS: ActRuntimeStatus | None = None
+
+
+def _get_act_runtime_status() -> ActRuntimeStatus:
+    """Return the runtime probe result, probing lazily on first call.
+
+    Probing is deferred so that tests that modify ACT, DOCKER_HOST
+    or related environment variables see the updated configuration.
+    """
+    global _ACT_RUNTIME_STATUS
```

Ruff (0.15.14)

[warning] Using the global statement to update `_ACT_RUNTIME_STATUS` is discouraged

(PLW0603)